drivers/serial/uart_stm32.c

```diff
@@ -1579,6 +1579,8 @@ static int uart_stm32_async_rx_enable(const struct device *dev,
 		return -EFAULT;
 	}
 
+	LL_USART_RequestRxDataFlush(config->usart);
```

Comment:
Not all STM32 chips has this function. You need probably `#ifdef USART_RQR_RXFRQ`. I guess  `LL_USART_ClearFlag_RXNE` is required in the `#else` part.

Reply:
Well, `LL_USART_ReceiveData8` also fixes the issue, is that available to all chips?

Reply:
I think so, you can check that in hal_stm32. But I don't know what happens when the UART has FIFO.Why don't you want to use `LL_USART_ClearFlag_RXNE` and `LL_USART_RequestRxDataFlush`?

Reply:
I want to keep it simple :) introducing ifdefs (especially when not necessary) is the root of all evil :D

Reply:
Agree, however it may be necessary here, because of FIFO (I'm not able to verify it). Anyway, #ifdef to flush the RX buffer is already done here https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/serial/uart_stm32.c#L1269. I suspect it is done this way for some reason.

Reply:
ok, lets do that :)

Reply:
LGTM.